src/array/cpu/labor_pick.h

```diff
@@ -24,6 +24,7 @@
 
 #include <dgl/array.h>
 #include <dgl/random.h>
+#include <dgl/pcg32.h>
```

Comment:
Is it possible to use mt19937 in this code?

Reply:
It would make it much slower, and results wouldn't be theoretically guaranteed to be correct either. That code doesn't use the PRNG in the usual way, which is to seed it once and get numbers. What I do is to keep reseeding before getting a new number.

Reply:
Thanks for educating me on this, do we have any measured results to show how much it is slower?
Also, why a different prng could make it "not theoretically guaranteed to be correct"? Conceptually they are just simulating random number.

If indeed we see the performance different, yes, we should switch to pcg.
Can you revert back to your original change, which is using pcg_random.hpp in include/dgl/random.h. And I am wondering whether you can include include/dgl/random.h in src/array/cpu/labor_pick.h, instead of including pcg32.h / pcg_random.hpp.

Again, thank you for your great and in-depth contribution.

Reply:
Usually, PRNG's are designed so that when you seed them once, every random number you sample after are independently identically distributed. But in my case, I require that the random numbers one gets by seeding with a different seed, every first random number you sample are independently identically distributed.

Reply:
In labor_pick.h, replacing
`pcg32(random_seed, t)` line with
```
std::seed_seq seq{(uint32_t)(random_seed >> 32), (uint32_t)random_seed, (uint32_t)(t >> 32), (uint32_t)t};
std::mt19937 ng(seq);
```
Causes 1000x slowdown.

Reply:
Can't even inspect if results are correct because the code is not terminating, as in comparing to the provided numbers in the original issue #4785.

Reply:
The reason why it is so slow is that `sizeof(std::mt19937)` returns 5000, so each time it needs to seed, it has to reinitialize a space of kilobytes as opposed to just two numbers as in pcg32.

Reply:
Also, I haven't run the benchmarks for it but I am certainly sure that the neighbor sampling cpu kernel will become at least 2x slower if RandomEngine switches from std::default_random_engine to std::mt19937/_64.

Reply:
One still needs to run benchmarks to see how much slower NeighborSampler will become when std::default_random_engine is swapped for pcg32.